addons/pos_viva_wallet/static/src/app/payment_viva_wallet.js

```diff
@@ -0,0 +1,199 @@
+/** @odoo-module */
+
+import { _t } from "@web/core/l10n/translation";
+import { PaymentInterface } from "@point_of_sale/app/payment/payment_interface";
+import { ErrorPopup } from "@point_of_sale/app/errors/popups/error_popup";
+import { sprintf } from "@web/core/utils/strings";
+import { uuidv4 } from "@point_of_sale/utils";
+
+export class PaymentVivaWallet extends PaymentInterface {
+
+    /*
+     Developer documentation:
+    https://developer.vivawallet.com/apis-for-point-of-sale/card-terminals-devices/rest-api/eft-pos-api-documentation/
+    */
+
+    setup() {
+        super.setup(...arguments);
+        this.paymentLineResolvers = {};
+    }
+    send_payment_request(cid) {
+        super.send_payment_request(cid);
+        return this._viva_wallet_pay(cid);
+    }
+    send_payment_cancel(order, cid) {
+        super.send_payment_cancel(order, cid);
+        return this._viva_wallet_cancel();
+    }
+    set_most_recent_service_id(id) {
+        this.most_recent_service_id = id;
+    }
+    pending_viva_wallet_line() {
+        return this.pos.getPendingPaymentLine("viva_wallet");
+    }
+
+    _handle_odoo_connection_failure(data = {}) {
+        // handle timeout
+        var line = this.pending_viva_wallet_line();
+        if (line) {
+            line.set_payment_status("retry");
+        }
+        this._show_error(
+            _t(
+                "Could not connect to the Odoo server, please check your internet connection and try again."
+            )
+        );
+
+        return Promise.reject(data); // prevent subsequent onFullFilled's from being called
+    }
+
+    _call_viva_wallet(data) {
+        return this.env.services.orm.silent
+            .call("pos.payment.method",
+                "viva_wallet_send_payment_request",
+                [[this.payment_method.id], data]
+            )
+            .catch(this._handle_odoo_connection_failure.bind(this));
+    }
+
+    _viva_wallet_handle_response(response) {
+        var line = this.pending_viva_wallet_line();
+        line.set_payment_status("waitingCard");
+        return this.waitForPaymentConfirmation();
+    }
+
+    _viva_wallet_pay () {
```

Comment:
Since you are already using async method, you can also convert this one to async. You'll have to adapt the return value.